prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -1812,15 +1804,19 @@ layoutWadlerLeijen
           | fits pWidth minNestingLevel availableWidth x -> x
           where
             minNestingLevel =
-                -- See https://github.com/quchen/prettyprinter/issues/83.
-                if startsWithLine y
-                    -- y might be a (more compact) hanging layout. Let's check x
-                    -- thoroughly with the smaller lineIndent.
-                    then lineIndent
-                    -- y definitely isn't a hanging layout. Let's allow the first
-                    -- line of x to be checked on its own and format it consistently
-                    -- with subsequent lines with the same indentation.
-                    else currentColumn
+                -- See the Note
+                -- [Choosing the right minNestingLevel for consistent smart layouts]
+                case y of
+                    SLine i _ ->
```

Comment:
Note that I don't try to look behind annotations anymore, mostly because it seemed unlikely to be useful. `dhall` doesn't need it in any case.

Reply:
I think you definitely should look behind annotations! Dhall is just one application of Prettyprinter, and the design of the latter should not be simplified because a revdep does not need it. Performance-wise it’s not an issue anyway (I wouldn’t be surprised if the check for `Ann` would be free here, actually).

Reply:
Fixed.